### ISSUE TITLE
Simplify on arrays does not account for constant arrays

### ISSUE THREAD

**Issue by marcogario** [pysmt/pysmt#599]
We have a concept of Array constant, and we have a way to manipulate it after reading a model etc.
However, the simplifier does not seem to be using it. In particular, if we have an array to which we perform a series of stores, the simplifier should compress it into a constant array. This would allow successive calls to Select to be simplified to the underlying value.

```
from pysmt.shortcuts import Symbol, Store, Select, Int, get_model
from pysmt.typing import ArrayType, INT

a = Symbol("a", ArrayType(INT, INT))
for i in range(10):
    a = Store(a, Int(i), Int(i*2))
x = Select(a, Int(5))
print(x.simplify().serialize())
# >>> a[0 := 0][1 := 2][2 := 4][3 := 6][4 := 8][5 := 10][6 := 12][7 := 14][8 := 16][9 := 18][5]
```
The print should instead output 10.

See https://stackoverflow.com/questions/57128894/how-to-use-array-in-pysmt

**Comment by marcogario** [comment]
The problem seems to be that even if `a` is a symbol and not a constant array. We do have the simplifications for handling the slightly different case: ```a = Array(INT, Int(0))```.

Arguable whether we should try to "lookahead" when simplifying a "Select" expression and see whether we can compute the value of the Select statically.

**Comment by mikand** [comment]
Exactly. Currently our simplifications are local, doing the simplification of your example would require the colletion of all the "cells" that we are sure of in a map (either on-demand or for every array-typed expression encountered) and then using such a map to simplify the selects.

I am unsure how useful such a simplification woould be, but it is doable.
